app/models/profile.rb

```diff
@@ -51,7 +61,16 @@ def self.mapped_attributes
   # the profiles table doesn't exist yet (e.g. when running bin/setup in a new
   # clone). I wish Rails had a hook for code to run after the app started, but
   # for now this is the best I can come up with.
-  refresh_attributes! if ApplicationRecord.connection.table_exists?("profiles")
+  def self.db_ready?
```

Comment:
This is used in `refresh_attributes`, which is necessary so we can always start the app with profiles that are aware of all their fields. We need this helper so the method can do an early return in scenarios where the DB does not exist yet, e.g. `bin/setup`.

The `begin` - `rescue` - `else` construct is a bit odd, but works well for our use case. There's no way to check for an AR connection except for actually calling it. If there is no DB, it will raise an `ActiveRecord::NoDatabaseError`, which we resuce. If we don't we execute the `else` branch, which ensures that the `profiles` table exists.

I really wish Rails had a callback that I could hook into for this, but for the time being this will have to do.